apollo-compiler/src/test/graphql/com/example/HeroNameDirective.json

```diff
@@ -0,0 +1,30 @@
+{
+  "operations": [
+    {
+      "operationName": "HeroNameDirective",
+      "operationType": "query",
+      "variables": [],
+      "source": "query HeroNameDirective {\n  hero {\n    __typename\n    name @include(if: false)\n  }\n}",
+      "fields": [
+        {
+          "responseName": "hero",
+          "fieldName": "hero",
+          "type": "Character",
+          "fields": [
+            {
+              "responseName": "name",
+              "fieldName": "name",
+              "type": "String!",
+              "isConditional": true
```

Comment:
From my understanding, it's not the client's responsibility to evaluate the conditional. 
The GraphQL execution engine will evaluate the conditional and decide whether or not to include the field in the result. 

So, I assume all the client needs to know is that the field is nullable.